core/src/main/java/bisq/core/xmr/connection/MoneroConnectionsManager.java

```diff
@@ -0,0 +1,181 @@
+package bisq.core.xmr.connection;
+
+import bisq.core.api.model.UriConnection;
+import bisq.core.btc.setup.WalletConfig;
+import bisq.core.xmr.connection.persistence.model.XmrConnectionList;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+
+import monero.common.MoneroConnectionManager;
+import monero.common.MoneroRpcConnection;
+
+@Slf4j
+@Singleton
+public final class MoneroConnectionsManager {
+
+    private final Object lock = new Object();
+
+    private final MoneroConnectionManager connectionManager;
+    private final XmrConnectionList connectionList;
+
+    @Inject
+    public MoneroConnectionsManager(MoneroConnectionManager connectionManager,
+                                    XmrConnectionList connectionList) {
+        this.connectionManager = connectionManager;
+        this.connectionList = connectionList;
+        initialize();
+    }
+
+    private void initialize() {
+        synchronized (lock) {
+            loadConnections();
+            addDefaultConnection();
+        }
+    }
+
+    private void loadConnections() {
+        connectionList.getConnections().forEach(uriConnection ->
+                connectionManager.addConnection(toMoneroRpcConnection(uriConnection)));
+    }
+
+    private void addDefaultConnection() {
+        String defaultUri = WalletConfig.MONERO_DAEMON_URI;
+        if (!connectionList.hasConnection(defaultUri)) {
+            addConnection(UriConnection.builder()
+                    .uri(defaultUri)
+                    .username(WalletConfig.MONERO_DAEMON_USERNAME)
```

Comment:
Getting a NPE if the username and password are null here. These fields should be nullable instead of expecting empty strings.

```
Jan-06 17:35:08.480 [HavenoDaemonMain] ERROR b.c.p.PersistenceManager: Error in saveToFile toProtoMessage: XmrConnectionList, XmrConnectionList 
java.lang.NullPointerException
	at protobuf.EncryptedUriConnection$Builder.setUsername(EncryptedUriConnection.java:704)
	at bisq.core.api.model.EncryptedUriConnection.toProtoMessage(EncryptedUriConnection.java:25)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1675)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at bisq.core.xmr.connection.persistence.model.XmrConnectionList.toProtoMessage(XmrConnectionList.java:297)
	at bisq.common.proto.persistable.PersistableEnvelope.toPersistableMessage(PersistableEnvelope.java:30)
	at bisq.common.persistence.PersistenceManager.persistNow(PersistenceManager.java:412)
	at bisq.common.persistence.PersistenceManager.lambda$maybeStartTimerForPersistence$8(PersistenceManager.java:401)
	at bisq.common.FrameRateTimer.run(FrameRateTimer.java:50)
	at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:807)
	at java.base/java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425)
	at bisq.common.MasterTimer$1.lambda$run$0(MasterTimer.java:38)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Nevermind. It appears protobuf does not like null being used directly in the model, so this is not an issue.

I'm working on the review and test.